### PR TITLE
repart: Allow specifying --copy-from more than once

### PR DESCRIPTION
**Pull request by daandemeyer**
Definitions will be synthesized from each of the given images.

### CODE REVIEW

**Comment by bluca** [comment]
Can you add some test coverage for this please?

**Comment by DaanDeMeyer** [comment]
@bluca We already have full coverage with the existing test. This just executes the exact same logic if you specify more than two sources.

**Comment by bluca** [comment]
I know, but having a sanity check to ensure it works correctly when reading from 2 images would be nice, no? For ASAN coverage and peace of mind

**Comment by poettering** [comment]
lgtm. thanks for addressing this so quickly

**Comment by DaanDeMeyer** [comment]
Updated integration test and added a trivial extra commit to always acquire a shared lock on the nodes passed to --copy-from as we never write to these

File: test/units/testsuite-58.sh [hunk 1]
```diff
@@ -160,12 +160,24 @@ last-lba: 2097118
...
+    sfdisk -d "$imgs/qqq" | grep -v -e 'sector-size' -e '^$'
+
     systemd-repart --offline="$OFFLINE" \
                    --empty=create \
                    --size=1G \
                    --dry-run=no \
                    --seed="$seed" \
-                   --copy-from="$imgs/zzz" \
+                   --definitions "" \
+                   --copy-from="$imgs/qqq" \
```
Comment (bluca) [inline test/units/testsuite-58.sh:179]: zzz?

Comment (daandemeyer) [reply test/units/testsuite-58.sh:179]: Nope qqq, zzz is too big

Comment (bluca) [reply test/units/testsuite-58.sh:179]: but then it's the same one used twice?

Comment (daandemeyer) [reply test/units/testsuite-58.sh:179]: Yeah, we just copy.the partitions from that one twice, that exercises the logic two times.

**Pull request merged**